### DISCUSSION TITLE
CLI能力与GUI能力的剥离规划

### DISCUSSION THREAD

**Discussion by anycodes** [Serverless-Devs/Serverless-Devs#42] [General]
希望CLI可以是更加轻量化的一个工具，不再提供GUI和应用中心相关操作，CLI单纯作为一个Serverless CLI Framework。

改动内容主要包括：
1.  指令的修改
2.  逻辑变更
3. 源的引入

# 指令的修改

指令的修改包括：
- init指令：增加自定义源配置
    当用户使用init的时候，系统会优先检测是不是一个地址，如果是github地址，则会init，如果不是github地址，则会判断是否有源，如果没有则会提醒配置源，如果有则会使用源
- search指令：增加自定义源配置
    会判断是否有源，如果没有则会提醒配置源，如果有则会使用源来进行search
- platform指令： 完全移除
- gui指令：完全移除
- set指令：增加一个配置源的能力
    s set shub

# 逻辑变更
- 组件下载逻辑：增加自定义源配置
    优先源，再使用github
- 插件下载逻辑：增加自定义源配置
    优先源，再使用github
- 获取help的帮助逻辑：先下载组件，并加载Publish.yaml

**Comment by rsonghuster** [comment]
源的引入：
- 源这块有什么具体的规划吗？ 还是说暂时只是定义了 github？ 那我们之前应用中心算是其中一个源？
- 如果做全球的话， 源都要加速功能吧， 各种语言都有国内版的加速， 比如淘宝 npm 源， pip 清华源 ...

**Reply by anycodes** [reply]
目前的规划是，默认是github，然后用户可以增加配置一个源。
如果用户是用过GUI来用的，系统会默认帮助用户配置一个我们的应用中心的源。
并且我们的应用中心会更名：serverless hub。
